monitoring/resourse_lakehouse_monitor.go

```diff
@@ -0,0 +1,139 @@
+package monitoring
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func removeSnapshotField(originalSchema map[string]*schema.Schema) map[string]*schema.Schema {
+	tmpSchema := make(map[string]*schema.Schema)
+	for k, v := range originalSchema {
+		tmpSchema[k] = v
+	}
+	delete(tmpSchema, "snapshot")
+	return tmpSchema
+}
+
+type Snapshot struct {
+}
+type Monitor struct {
+	AssetsDir                string                                   `json:"assets_dir"`
+	BaselineTableName        string                                   `json:"baseline_table_name,omitempty"`
+	CustomMetrics            []catalog.MonitorCustomMetric            `json:"custom_metrics,omitempty"`
+	DataClassificationConfig *catalog.MonitorDataClassificationConfig `json:"data_classification_config,omitempty"`
+	FullName                 string                                   `json:"table_name"`
+	InferenceLog             *catalog.MonitorInferenceLogProfileType  `json:"inference_log,omitempty"`
+	Notifications            []catalog.MonitorNotificationsConfig     `json:"notifications,omitempty"`
+	OutputSchemaName         string                                   `json:"output_schema_name"`
+	Schedule                 *catalog.MonitorCronSchedule             `json:"schedule,omitempty"`
+	Snapshot                 Snapshot                                 `json:"snapshot,omitempty"`
+	SkipBuiltinDashboard     bool                                     `json:"skip_builtin_dashboard,omitempty"`
+	SlicingExprs             []string                                 `json:"slicing_exprs,omitempty"`
+	TimeSeries               *catalog.MonitorTimeSeriesProfileType    `json:"time_series,omitempty"`
+	WarehouseId              string                                   `json:"warehouse_id,omitempty"`
+}
+
+type MonitorInfo struct {
+	Monitor
+	DriftMetricsTableName   string                    `json:"drift_metrics_table_name,omitempty"`
+	ProfileMetricsTableName string                    `json:"profile_metrics_table_name,omitempty"`
+	Status                  catalog.MonitorInfoStatus `json:"status,omitempty"`
```

Comment:
This also should be a pointer

Reply:
MonitorInfoStatus is just a string enum. If I make it a pointer, the `StructToSchema` method expects it to be a struct.